### DISCUSSION TITLE
Some edge functions returns a random 405 error

### DISCUSSION THREAD

**Discussion by mattiaz9** [vercel/community#3475] [Help]
### Summary

In the past few days some edge functions (which are POST requests) stopped working returning a 405 error.

This is weird because is not a `Method not allowed` problem. In fact, looking at the logs, it turned out to be a 500 internal error of the edge function:
```bash
[POST] /api/documents/save?nxtPparams=documents%2Fsave
reason=INTERNAL_EDGE_FUNCTION_UNHANDLED_ERROR, status=500, upstream_status=500, user_error=false
```

It seems to be related to only edge runtime, because switching to `runtime = 'nodejs'` fixed the issue.

I also put some logs inside the POST handler and none of those were printed, probably meaning the endpoint was never called in the first place.

### Steps to Reproduce

It's not easy reproducible. It seems to happen only for POST request on the edge runtime, but not every one have this issue.

**Comment by robinrogowski** [comment]
This happens on our implementation as well. Keep us posted if you find anything.

**Comment by amyegan** [comment]
There are some [Edge Functions limitations](https://vercel.com/docs/concepts/functions/edge-functions/limitations#) that could be causing errors. A common one is that many Node.js APIs are not available. You can find a more complete list in the Edge Function docs at https://vercel.com/docs/concepts/functions/edge-functions/limitations#

I also found a few other posts about 405 errors which could be related to your situation:
- https://github.com/vercel/next.js/discussions/51984
- https://github.com/vercel/community/discussions/3081
- https://github.com/vercel/community/discussions/3138

**Reply by mattiaz9** [reply]
I don't think that is the case. 

The local environment should simulate the edge runtime APIs, so it should also crash locally, which is not the case with my issue.
Also my endpoint is a catch all segment, meaning there are many other endpoints that works just fine that are in the same bundled function.

The weird thing is that it doesn't even enter the POST handler.

**Comment by poweroutlet2** [comment]
Did you ever find the cause of this or a solution? Running into the same problem

**Reply by mattiaz9** [reply]
nope! my guess it might be some internal edge runtime bug. For the moment I have to stick with nodejs.

**Comment by Yidadaa** [comment]
I also encountered the same problem, which is very strange. It seems to be caused by the unstable operation of the Edge Runtime, and there is no Error Log to trace back to!

**Comment by shardbearer-shri** [comment]
Has anyone found a solution to this? Iam still facing this issue with the edge runtime

**Comment by mdatsev** [comment]
I was getting the same random 405 responses and I was not even using edge runtime, just plain node runtime.

What I noticed though is that I had used up 94% of my free plan bandwidth.

Once I upgraded to pro and redeployed the errors went away.

**Comment by ymc9** [comment]
Same here, getting random 405 response on edge runtime with next.js and vercel postgres.

reason=INTERNAL_EDGE_FUNCTION_INVOCATION_FAILED, status=500, upstream_status=500, user_error=false

**Reply by amyegan** [reply]
What version of Next.js does your project use?

**Reply by ymc9** [reply]
I'm using Next.js 14.1.4

**Comment by robinrogowski** [comment]
We recently upgraded to Next 13.2.0 to start a migration by using the app router for some API routes that we have.
________________________________
